lib/model.js

```diff
@@ -3831,8 +3832,21 @@ class Model {
                   const values = {};
                   values[include.association.foreignKey] = this.get(this.constructor.primaryKeyAttribute, {raw: true});
                   values[include.association.otherKey] = instance.get(instance.constructor.primaryKeyAttribute, {raw: true});
-                  // Include values defined in the scope of the association
+                  
+                  // Include values defined in the association
                   Object.assign(values, include.association.through.scope);
+                  if (instance[include.association.through.model.name]) {
+                    for (const attr of Object.keys(include.association.through.model.rawAttributes)) {
+                      if (include.association.through.model.rawAttributes[attr]._autoGenerated) {
+                        continue;
+                      }
+                      if (attr === include.association.foreignKey || attr === include.association.otherKey) {
+                        continue;
+                      }
+                      values[attr] = instance[include.association.through.model.name][attr];
```

Comment:
Check if `instance[include.association.through.model.name][attr]` is defined only then we should set `values[attr]`